test/configuration/vimrcKeyRemappingBuilder.test.ts

```diff
@@ -0,0 +1,64 @@
+import * as assert from 'assert';
+import { IKeyRemapping, IVimrcKeyRemapping } from '../../src/configuration/iconfiguration';
+import { vimrcKeyRemappingBuilder } from '../../src/configuration/vimrcKeyRemappingBuilder';
+
+suite('VimrcKeyRemappingBuilder', () => {
+  test('Build IKeyRemapping objects from .vimrc lines', () => {
+    const testCases = [
+      {
+        vimrcLine: 'nnoremap <C-h> <<',
+        keyRemapping: {
+          before: ['<C-h>'],
+          after: ['<', '<'],
+        },
+        keyRemappingType: 'nnoremap',
+        expectNull: false
+      },
+      {
+        vimrcLine: 'imap jj <Esc>',
+        keyRemapping: {
+          before: ['j', 'j'],
+          after: ['<Esc>'],
+        },
+        keyRemappingType: 'imap',
+        expectNull: false
+      },
+      {
+        vimrcLine: 'vnoremap <leader>" c""<Esc>P',
+        keyRemapping: {
+          before: ['<leader>', '"'],
+          after: ['c', '"', '"', '<Esc>', 'P'],
+        },
+        keyRemappingType: 'vnoremap',
+        expectNull: false
+      },
+      {
+        // Mapping with a command
+        vimrcLine: 'nnoremap <C-s> :w',
+        keyRemapping: {
+          before: ['<C-s>'],
+          commands: [':w'],
+        },
+        keyRemappingType: 'nnoremap',
+        expectNull: false
+      },
+      {
+        // Ignore non-mapping lines
+        vimrcLine: 'set scrolloff=8',
```

Comment:
Since you opened this PR, `scrolloff` has been introduced as `editor.cursorSurroundingLines`

Reply:
Resolving this because it's out of scope for the PR